apps/content-publishing-api/src/controllers/v3/content.controller.v3.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  Controller,
+  HttpCode,
+  Post,
+  Inject,
+  HttpStatus,
+  Req,
+  Res,
+  Logger,
+  BadRequestException,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import { ApiOperation, ApiResponse, ApiTags, ApiConsumes, ApiBody } from '@nestjs/swagger';
+import { Request, Response } from 'express';
+import Busboy from 'busboy';
+import { FilesUploadDto, BatchAnnouncementResponseDto } from '#types/dtos/content-publishing';
+import { ApiService } from '../../api.service';
+import apiConfig, { IContentPublishingApiConfig } from '#content-publishing-api/api.config';
+import { SkipInterceptors } from '#utils/decorators/skip-interceptors.decorator';
+import { IFileResponse } from '#types/interfaces/content-publishing';
+
+@Controller({ version: '3', path: 'content' })
+@ApiTags('v3/content')
+export class ContentControllerV3 {
+  private readonly logger: Logger;
+
+  constructor(
+    @Inject(apiConfig.KEY) private readonly appConfig: IContentPublishingApiConfig,
+    private readonly apiService: ApiService,
+  ) {
+    this.logger = new Logger(this.constructor.name);
+  }
+
+  @Post('uploadBatchAnnouncement')
+  @SkipInterceptors()
+  @HttpCode(HttpStatus.ACCEPTED)
+  @ApiConsumes('multipart/form-data')
+  @ApiOperation({
+    summary: 'Upload files and create batch announcements',
+    description:
+      'Upload files with their corresponding schema IDs to create batch announcements. Each file must be accompanied by a schemaId field.',
+  })
+  @ApiBody({
+    description: 'Asset files',
+    type: FilesUploadDto,
+  })
+  @ApiResponse({ status: '2XX', type: BatchAnnouncementResponseDto })
+  async postUploadBatchAnnouncement(
```

Comment:
Q: since the file upload logic is the same as `POST /v2/asset/upload` (a great example of NOT following the above-mentioned REST endpoint-naming pattern--but we chose consistency with v1, which was implemented before we started following that standard), I wonder if any of the file upload handling could be refactored to a common function?